src/Catel.Core/Catel.Core.Shared/Data/ModelBase.cs

```diff
@@ -626,6 +642,11 @@ protected void SetParent(IParent parent)
         /// </exception>
         public override bool Equals(object obj)
         {
+            if (DisableUsageOfEqualityComparerForEquals)
+            {
+                return RuntimeHelpers.Equals(this, obj);
```

Comment:
Awesome, thanks for verifying. Note that I started adding performance tests to Catel. If you have something to share, we can use this and benchmark for future builds: https://github.com/Catel/Catel.Benchmarks